Firestore/Example/Tests/Integration/API/FIRServerTimestampTests.m

```diff
@@ -88,14 +103,46 @@ - (void)writeInitialData {
   XCTAssertEqualObjects(initialDataSnap.data, _initialData);
 }
 
+/** Waits for a snapshot with local writes. */
+- (FIRDocumentSnapshot *)waitForLocalEvent {
+  return [_accumulator awaitEventWithName:@"Local event."];
+}
+
 /** Waits for a snapshot containing _setData but with NSNull for the timestamps. */
-- (void)waitForLocalEvent {
-  FIRDocumentSnapshot *localSnap = [_accumulator awaitEventWithName:@"Local event."];
+- (FIRDocumentSnapshot *)waitForLocalEventWithNullTimestamps {
+  FIRDocumentSnapshot *localSnap = [self waitForLocalEvent];
   XCTAssertEqualObjects(localSnap.data, [self expectedDataWithTimestamp:[NSNull null]]);
+  return localSnap;
+}
+
+/** Waits for a snapshot containing _setData but with a local estimate for the timestamps. */
+- (FIRDocumentSnapshot *)waitForLocalEventWithEstimatedTimestamps {
```

Comment:
It doesn't seem like the concerns of waiting for a local event and validating the values within the snapshot in the event are necessarily conjoined. If you split this you could just have a `verifyEstimatedTimestampsInSnapshot:` method and just one `waitForLocalEvent` method.

Similarly below regarding the validation of the previous value.

This would turn the call sites in the individual tests into multiple lines but it would keep the concerns of sequencing separate from verification.

Reply:
I split this and the existing methods up and changed all tests.